### PR TITLE
update Azure AD docs

### PR DESCRIPTION
**Pull request by jmyers82**
### What
Update Azure AD setup docs for configuring SAML MemberOf claim

### Why
Current docs are out of date. 

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [x] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [x] Description links to related pull requests or issues, if any.

#### Content
- [x] Redirects have been added for moved, renamed, or deleted pages. This requires a separate PR in the [`terraform-website` repository](https://github.com/hashicorp/terraform-website) `redirects.next.js` file.
- [x] API documentation and the API Changelog have been updated. 
- [x] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [x] Pages with related content are updated and link to this content when appropriate.
- [x] Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [x] New pages have metadata (page name and description) at the top.
- [x] New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [x] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [x] UI elements (button names, page names, etc.) are bolded.
- [x] The Vercel website preview successfully deployed.

#### Reviews
- [x] I or someone else reviewed the content for technical accuracy.
- [x] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

**Review by mgarrell777 — APPROVED** [review]
Hi @jmyers82 I'm Matthew, new technical writer with the Terraform team. Very good stuff here. I approve with the 1 small rewording change suggestion.

Thanks very much for your time and effort.

File: website/docs/cloud-docs/users-teams-organizations/single-sign-on/azure-ad.mdx [hunk 1]
```diff
@@ -79,15 +79,22 @@ To edit your Azure SSO configuration settings:
...
-   We recommend naming it "MemberOf", leaving the namespace blank, and potentially sourcing `user.assignedroles` as an easy starting point.
+1. Edit step 2, "User Attributes & Claims."  
+2. Add a new group claim.
+3. In the "Group Claims" side window, select the "Security Groups" radio button.  
+   For "Source Attribute", select `sAMAccountName` to have group names passed in the SAML or select "Group ID" to pass thru the UUIDs for the group.  See **Note** below for more information on UUID's.  
+   Check the box for "Customize the name of the group claim" and in the "Name (required)" field put "MemberOf" and leave namespace blank.
 
 -> **Note:** When Azure AD is configured to use Group Claims, it provides Group UUIDs instead of human readable names in its SAML assertions. We recommend [configuring SSO Team IDs](/cloud-docs/users-teams-organizations/single-sign-on#team-names-and-sso-team-ids) for your Terraform Cloud teams to match these Azure Group UUIDs.
 
 If you plan to make use of SAML to set usernames in your Microsoft Azure AD application:
```
Comment (mgarrell777) [inline website/docs/cloud-docs/users-teams-organizations/single-sign-on/azure-ad.mdx:90]: Suggest rewording to

"If you plan to use SAML"

File: website/docs/cloud-docs/users-teams-organizations/single-sign-on/azure-ad.mdx [hunk 2]
```diff
@@ -79,15 +79,22 @@ To edit your Azure SSO configuration settings:
 To configure team management in your Microsoft Azure AD application:
 
 1. Navigate to the single sign-on page.
-1. Edit step 2, "User Attributes & Claims."
-   We recommend naming it "MemberOf", leaving the namespace blank, and potentially sourcing `user.assignedroles` as an easy starting point.
+1. Edit step 2, "User Attributes & Claims."  
+2. Add a new group claim.
+3. In the "Group Claims" side window, select the "Security Groups" radio button.  
+   For "Source Attribute", select `sAMAccountName` to have group names passed in the SAML or select "Group ID" to pass thru the UUIDs for the group.  See **Note** below for more information on UUID's.
```
[the remark this answers is not in the document]
Comment (jmyers82) [reply website/docs/cloud-docs/users-teams-organizations/single-sign-on/azure-ad.mdx]: this has been updated and commited.

File: website/docs/cloud-docs/users-teams-organizations/single-sign-on/azure-ad.mdx (hunk 1 above)
Comment (jmyers82) [reply website/docs/cloud-docs/users-teams-organizations/single-sign-on/azure-ad.mdx]: This has been updated and committed.

File: website/docs/cloud-docs/users-teams-organizations/single-sign-on/azure-ad.mdx [hunk 3]
```diff
@@ -79,15 +79,22 @@ To edit your Azure SSO configuration settings:
...
 1. Navigate to the single sign-on page.
-1. Edit step 2, "User Attributes & Claims."
-   We recommend naming it "MemberOf", leaving the namespace blank, and potentially sourcing `user.assignedroles` as an easy starting point.
+1. Edit step 2, "User Attributes & Claims."  
+2. Add a new group claim.
+3. In the "Group Claims" side window, select the "Security Groups" radio button.  
+   For "Source Attribute", select `sAMAccountName` to have group names passed in the SAML or select "Group ID" to pass thru the UUIDs for the group.  See **Note** below for more information on UUID's.  
+   Check the box for "Customize the name of the group claim" and in the "Name (required)" field put "MemberOf" and leave namespace blank.
 
 -> **Note:** When Azure AD is configured to use Group Claims, it provides Group UUIDs instead of human readable names in its SAML assertions. We recommend [configuring SSO Team IDs](/cloud-docs/users-teams-organizations/single-sign-on#team-names-and-sso-team-ids) for your Terraform Cloud teams to match these Azure Group UUIDs.
```
[the remark this answers is not in the document]
Comment (jmyers82) [reply website/docs/cloud-docs/users-teams-organizations/single-sign-on/azure-ad.mdx]: This has been updated and committed.

File: website/docs/cloud-docs/users-teams-organizations/single-sign-on/azure-ad.mdx [hunk 4]
```diff
@@ -79,15 +79,22 @@ To edit your Azure SSO configuration settings:
 To configure team management in your Microsoft Azure AD application:
 
 1. Navigate to the single sign-on page.
-1. Edit step 2, "User Attributes & Claims."
-   We recommend naming it "MemberOf", leaving the namespace blank, and potentially sourcing `user.assignedroles` as an easy starting point.
+1. Edit step 2, "User Attributes & Claims."  
+2. Add a new group claim.
+3. In the "Group Claims" side window, select the "Security Groups" radio button.  
+   In the "Source Attribute" field, select either sAMAccountName to pass group names in the SAML or "Group ID" to pass thru the UUIDs for the group. See the related Note for more information on UUIDs.
+   Check the "Customize the name of the group claim" box. Use "MemberOf" in the "Name (required)" field. Leave the namespace field blank.
```
Comment (laurapacilio) [inline website/docs/cloud-docs/users-teams-organizations/single-sign-on/azure-ad.mdx:86]: A couple of questions for you:
- Is this the right name for this field? sAMAccountName. (Part of me felt like it could have been SAML, so I just wanted to make sure).
- I felt a bit confused by the phrase "pass through the UUIDs". I changed it to "use" because I think we're specifying what attribute to use, but please let me know if I misinterpreted that!
- Does the user need to click Save or anything after all of this is done? I just want to make sure this is the final step of the process.

Finally, just a couple of style nits that I fixed in my suggestion:
- We always want to use "Refer to" instead of "See", per our style guide.
- I added a few more numbered steps to break up the instructions a bit. We want each new action in the process to be a numbered step.
- We always want to use 1 for numbered steps (the website will automatically know which number to use next, and this helps make sure we don't end up going from like 1 to 3 by accident when we make edits in the future!).
- All UI elements (button names, etc.) should be in bold.
- We can just say the name of the button, so "Click **Enter**" rather than "Click the **Enter** button.

Let me know what you think of these edits/my questions. Thank you so much for fixing this for us!

```suggestion
1. Edit step 2, **User Attributes & Claims.** 
1. Add a new group claim.
1. In **Group Claims**, select **Security Groups**.  
1. In the **Source Attribute** field, select either **sAMAccountName** to use account names or **Group ID** to use group UUIDs. 
1. Check **Customize the name of the group claim**. 
1. Set **Name (required)** to "MemberOf" and leave the namespace field blank.
```

File: website/docs/cloud-docs/users-teams-organizations/single-sign-on/azure-ad.mdx [hunk 5]
```diff
@@ -79,15 +79,22 @@ To edit your Azure SSO configuration settings:
...
 
--> **Note:** When Azure AD is configured to use Group Claims, it provides Group UUIDs instead of human readable names in its SAML assertions. We recommend [configuring SSO Team IDs](/cloud-docs/users-teams-organizations/single-sign-on#team-names-and-sso-team-ids) for your Terraform Cloud teams to match these Azure Group UUIDs.
+-> **Note:** If you configure Azure AD to use Group Claims, it provides Group UUIDs instead of human readable names in its SAML assertions. We recommend [configuring SSO Team IDs](/cloud-docs/users-teams-organizations/single-sign-on#team-names-and-sso-team-ids) for your Terraform Cloud teams to match these Azure Group UUIDs.
```
Comment (laurapacilio) [inline website/docs/cloud-docs/users-teams-organizations/single-sign-on/azure-ad.mdx:88]:
```suggestion
-> **Note:** When you configure Azure AD to use Group Claims, it provides Group UUIDs instead of human readable names in its SAML assertions. We recommend [configuring SSO Team IDs](/cloud-docs/users-teams-organizations/single-sign-on#team-names-and-sso-team-ids) for your Terraform Cloud teams to match these Azure Group UUIDs.
```

File: website/docs/cloud-docs/users-teams-organizations/single-sign-on/azure-ad.mdx [hunk 6]
```diff
@@ -79,15 +79,22 @@ To edit your Azure SSO configuration settings:
...
 
-If you plan to make use of SAML to set usernames in your Microsoft Azure AD application:
+If you plan to use SAML to set usernames in your Microsoft Azure AD application:
 
 1. Navigate to the single sign-on page.
-1. Edit step 2, "User Attributes & Claims."
+1. Edit step 2, "User Attributes & Claims."
```
Comment (laurapacilio) [inline website/docs/cloud-docs/users-teams-organizations/single-sign-on/azure-ad.mdx:93]:
```suggestion
1. Edit step 2, **User Attributes & Claims.**  
```

**Review by laurapacilio — CHANGES_REQUESTED** [review]
Hi @jmyers82  A couple more comments. :-)  Please let me know if my edits have materially changed or misinterpreted the meaning. Thank you so much for your work on this - we really appreciate it!

File: website/docs/cloud-docs/users-teams-organizations/single-sign-on/azure-ad.mdx [hunk 7]
```diff
@@ -80,15 +80,23 @@ To edit your Azure SSO configuration settings:
...
-   We recommend naming it "MemberOf", leaving the namespace blank, and potentially sourcing `user.groups` as an easy starting point.
+1. Edit step 2, **User Attributes & Claims**. 
+1. Add a new group claim.
+1. In **Group Claims**, select **Security Groups**.  
+1. In the **Source Attribute** field, select either **sAMAccountName** to use account names or **Group ID** to use group UUIDs. 
+1. Check **Customize the name of the group claim**. 
+1. Set **Name (required)** to "MemberOf" and leave the namespace field blank.
 
--> **Note:** When Azure AD is configured to use Group Claims, it provides Group UUIDs instead of human readable names in its SAML assertions. We recommend [configuring SSO Team IDs](/terraform/cloud-docs/users-teams-organizations/single-sign-on#team-names-and-sso-team-ids) for your HCP Terraform teams to match these Azure Group UUIDs.
+-> **Note:** When you configure Azure AD to use Group Claims, it provides Group UUIDs instead of human readable names in its SAML assertions. We recommend [configuring SSO Team IDs](/cloud-docs/users-teams-organizations/single-sign-on#team-names-and-sso-team-ids) for your HCP Terraform teams to match these Azure Group UUIDs.
```
Comment (rkoron007) [inline website/docs/cloud-docs/users-teams-organizations/single-sign-on/azure-ad.mdx:90]:
```suggestion
-> **Note:** When you configure Azure AD to use Group Claims, it provides Group UUIDs instead of human readable names in its SAML assertions. We recommend [configuring SSO Team IDs](/terraform/cloud-docs/users-teams-organizations/single-sign-on#team-names-and-sso-team-ids) for your HCP Terraform teams to match these Azure Group UUIDs.
```

File: website/docs/cloud-docs/users-teams-organizations/single-sign-on/azure-ad.mdx [hunk 8]
```diff
@@ -80,15 +80,23 @@ To edit your Azure SSO configuration settings:
...
 
-If you plan to make use of SAML to set usernames in your Microsoft Azure AD application:
+If you plan to use SAML to set usernames in your Microsoft Azure AD application:
 
 1. Navigate to the single sign-on page.
-1. Edit step 2, "User Attributes & Claims."
-   We recommend naming the claim "username", leaving the namespace blank, and sourcing `user.displayname` or `user.mailnickname` as a starting point. Note that HCP Terraform usernames only allow lowercase letters, numbers, and dashes.
+1. Edit step 2, **User Attributes & Claims.**  
+   We recommend naming the claim "Username", leaving the namespace blank, and sourcing something like `user.displayname` or `user.mailnickname`.  Note that HCP Terraform usernames only allow lowercase letters, numbers, and dashes.
```
Comment (rkoron007) [inline website/docs/cloud-docs/users-teams-organizations/single-sign-on/azure-ad.mdx:96]:
```suggestion
   We recommend naming the claim "username", leaving the namespace blank, and sourcing `user.displayname` or `user.mailnickname` as a starting point. Note that HCP Terraform usernames only allow lowercase letters, numbers, and dashes.
```

File: website/docs/cloud-docs/users-teams-organizations/single-sign-on/azure-ad.mdx [hunk 9]
```diff
@@ -80,15 +80,23 @@ To edit your Azure SSO configuration settings:
...
 
 1. Navigate to the single sign-on page.
-1. Edit step 2, "User Attributes & Claims."
+1. Edit step 2, **User Attributes & Claims.**  
    We recommend naming the claim "username", leaving the namespace blank, and sourcing `user.displayname` or `user.mailnickname` as a starting point. Note that HCP Terraform usernames only allow lowercase letters, numbers, and dashes.
 
 If you namespaced any of your claims, note that the attribute name passed by Microsoft Azure AD will follow the form `<claim_namespace/claim_name>`. Consider this when setting Team and Username attribute names.
+
+## Troubleshooting the SAML assertion
+Use this guide to verify and validate the claims being sent in the SAML response.
```
Comment (rkoron007) [inline website/docs/cloud-docs/users-teams-organizations/single-sign-on/azure-ad.mdx:101]:
```suggestion
[Use this guide](https://support.hashicorp.com/hc/en-us/articles/1500005371682-Capturing-a-SAML-Assertion) to verify and validate the claims being sent in the SAML response. 
```

File: website/docs/cloud-docs/users-teams-organizations/single-sign-on/azure-ad.mdx [hunk 10]
```diff
@@ -80,15 +80,23 @@ To edit your Azure SSO configuration settings:
...
 1. Navigate to the single sign-on page.
-1. Edit step 2, "User Attributes & Claims."
+1. Edit step 2, **User Attributes & Claims.**  
    We recommend naming the claim "username", leaving the namespace blank, and sourcing `user.displayname` or `user.mailnickname` as a starting point. Note that HCP Terraform usernames only allow lowercase letters, numbers, and dashes.
 
 If you namespaced any of your claims, note that the attribute name passed by Microsoft Azure AD will follow the form `<claim_namespace/claim_name>`. Consider this when setting Team and Username attribute names.
+
+## Troubleshooting the SAML assertion
+[Use this guide](https://support.hashicorp.com/hc/en-us/articles/1500005371682-Capturing-a-SAML-Assertion) to verify and validate the claims being sent in the SAML response. 
+https://support.hashicorp.com/hc/en-us/articles/1500005371682-Capturing-a-SAML-Assertion
```
Comment (rkoron007) [inline website/docs/cloud-docs/users-teams-organizations/single-sign-on/azure-ad.mdx:102]:
```suggestion
```

**Review by rkoron007 — APPROVED** [review]
Approving this and adding suggestions to get this merged!
